tests/test_clpir.py

```diff
@@ -276,5 +276,46 @@ def __check_log_event(
         self.assertEqual(idx, expected_idx, f"Message idx: {idx}, Expected: {expected_idx}")
 
 
+class TestCaseFourByteEncoder(unittest.TestCase):
+    """
+    Class for testing clp_ffi_py.CLPIR.FourByteEncoder.
+    """
+
+    def test_init(self) -> None:
+        type_error_exception_captured: bool = False
+        four_byte_encoder: FourByteEncoder
+        try:
+            four_byte_encoder = FourByteEncoder()  # noqa
+        except TypeError:
+            type_error_exception_captured = True
+        self.assertEqual(
+            type_error_exception_captured, True, "FourByteEncoder should be non-instantiable."
+        )
+
+    def test_encode(self) -> None:
+        """
+        For simplicity, this function is not testing the functionality of the
+        encoder.
+
+        It only checks if the result of
+        encode_message_and_timestamp_delta is consistent with the combine of
+        encode_message and encode_timestamp_delta. The actual functionality
+        should be covered by the unittest of CLP Python logging library.
+        TODO: When the decoder is implemented, add some more tests to ensure
+        the encoded bytes can be successfully decoded to recover the original
+        log event.
+        """
```

Comment:
I think we can just rename this test (and rewrite the doc string accordingly) to be explicitly for testing whether "`encode_message_and_timestamp_delta` is consistent with the addition of `encode_message` and `encode_timestamp_delta`".

We can move the TODO and comments on general encoding tests to the class doc string.